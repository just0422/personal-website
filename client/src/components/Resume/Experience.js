import React, {Component} from 'react';
import {Grid, Row, Col} from 'react-bootstrap';
import Moment from 'react-moment';
import axios from 'axios';

import api from 'APIUtils';
import ErrorModal from 'Error';

import 'stylesheets/resume.scss';

export default class Experience extends Component {
  constructor(props) {
    super(props);

    this.state = {
      skills: [],
      comments: [],
      error: null,
    };
  }

  static defaultProps = {
    job: {
      id: -1,
      name: '',
      title: '',
      start: '',
      end: '',
    },
  };

  componentDidMount() {
    let id = this.props.job.id;
    if (id > 0) {
      axios
				.all([
					api.experiences().getSkills(id),
          axios.get('/api/v1/experiences/' + id + '/comments'),
        ])
        .then(
          axios.spread((skillsResponse, commentsResponse) => {
            this.setState({
              skills: skillsResponse.data,
              comments: commentsResponse.data,
            });
          }),
        )
        .catch(err => {
          this.setState({
            error: err,
          });
        });
    }
  }

  render() {
    let job = this.props.job;
    let skills = this.state.skills
      .map(skill => {
        return skill.name;
      })
      .join();
    let comments = this.state.comments.map((comment, i) => {
      return <li key={i}>{comment.content}</li>;
		});

    if (this.state.error) {
		console.log("ErrorModal should render");
      return <ErrorModal component="Experience" error={this.state.error} />;
    } else {
      return (
        <div>
          <Grid>
            <Row>
              <Col xs={6}>
                <div className="section-element-header">
                  <strong>{job.name}</strong>
                </div>
              </Col>
              <Col xs={6} className="section-element-dates">
                <Moment format={'MMM. YYYY'}>{job.start}</Moment> -{' '}
                <Moment format={'MMM. YYYY'}>{job.end}</Moment>
              </Col>
              <Col xs={12} className="section-element-header">
                <em>
                  {job.title} ({skills})
                </em>
              </Col>
              <Col xs={12} className="section-element-comments">
                <ul>{comments}</ul>
              </Col>
            </Row>
          </Grid>
        </div>
      );
    }
  }
}
