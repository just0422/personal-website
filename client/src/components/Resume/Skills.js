import React, {Component} from 'react';
import {Grid, Row, Col} from 'react-bootstrap';

import 'stylesheets/resume.scss';

export default class Skills extends Component {
  render() {
    return (
      <div>
        <h3 className="resume-subheader">Skills</h3>
        <Grid>
          <Row>
            {this.props.skills.map((skill, i) => {
              return (
                <Col xs={1} key={i}>
                  {skill.name}
                </Col>
              );
            })}
          </Row>
        </Grid>
      </div>
    );
  }
}
