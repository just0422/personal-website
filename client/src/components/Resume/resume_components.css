.resume-subheader{
	text-align: left
}
