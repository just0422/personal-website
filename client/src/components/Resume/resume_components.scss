.resume-subheader{
	text-align: left;
}

.section-element-header {
	text-align: left;
	margin-left: 20px;
	width: 100%;
}

.section-element-details {
	text-align: left;
	padding-left: 60px;
}

.section-element-dates {
	text-align: right;
}

.section-element-comments {
	text-align: left;
	padding-left: 20px;
}
