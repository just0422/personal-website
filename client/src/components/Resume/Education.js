import React, { Component } from 'react';
import { Row, Col } from 'react-bootstrap';

import './resume_components.css';

export default class Education extends Component {
	render() {
		return (
			<div>
				<h3 className="resume-subheader">Education</h3>
				<div className="section-element-header">
					<strong>Stony Brook University, New York</strong>
				</div>
				<Row>
					<Col xs={6} className="section-element-details">M.S. Computer Science</Col>
					<Col xs={6} className="section-element-dates">Aug. 2016 - Aug. 2018</Col>
				</Row>
				<Row>
					<Col xs={6} className="section-element-details">B.S. Computer Science</Col>
					<Col xs={6} className="section-element-dates">Aug. 2012 - May 2016</Col>
				</Row>
			</div>
		);
	}
}
