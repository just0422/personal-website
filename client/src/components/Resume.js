import React, { Component } from 'react';
import { Grid, Row, Col, PageHeader } from 'react-bootstrap';
import axios from 'axios';
import {PacmanLoader} from 'react-spinners';

import Skills from 'Resume/Skills';
import Education from 'Resume/Education';
import Experience from 'Resume/Experience';
import api from 'APIUtils';

import ErrorModal from 'Error';

export default class Resume extends Component {
	constructor(props){
		super(props);

		this.state = {
			skills: [],
			experiences: [],
			error: null,
      loading: false,
		}
	}

	componentDidMount(){
    this.setState({loading: true});
		axios.all([
			api.skills().getAll(),
			api.experiences().getAll()
		]).then(axios.spread((skillsResponse, experiencesResponse) => {
        console.log(skillsResponse);
        console.log(experiencesResponse);
				this.setState({
          loading: false,
					skills: skillsResponse.data,
					experiences: experiencesResponse.data
				})
		})).catch(err => {
			this.setState({
        loading: false,
				error: err,
			})
		});
	}

	render(){
		if (this.state.error) {
			return <ErrorModal component="Resume" error={this.state.error} />
    } else if (this.state.loading) {
      return (
        <div className="container">
          <PageHeader>Justin Maldonado</PageHeader>
          <PacmanLoader
            sizeUnit={'px'}
            size={150}
            color={'#a00'}
            loading={this.state.loading}
            className="justify-content-center"
          />
        </div>
      );
		} else {
			return(
				<div className="container">
					<PageHeader>Justin Maldonado</PageHeader>
					<Grid>
						<Row>
							<Col xs={4}><a href="https://github.com/just0422" target="_blank" rel="noopener noreferrer">https://github.com/just0422</a></Col>
							<Col xs={4}><a href="mailto:just0422@gmail.com">just0422@gmail.com</a></Col>
							<Col xs={4}>(347) 922-5075</Col>
						</Row>
						<Row>
							<Col md={5} sm={12}>
								<Education />
							</Col>
							<Col md={7} sm={12}>
								<Skills skills={this.state.skills}/>
							</Col>
						</Row>
						<hr />
						<h3 className="resume-subheader">Experience</h3>
						{
							this.state.experiences.map( (job, i) => {
								return (<Experience job={job} key={i}/>)
							})
						}
					</Grid>
					<br />
					<br />
				</div>
			);
		}
	}
}
