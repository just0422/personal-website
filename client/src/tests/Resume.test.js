import React from 'react';
import {shallow, mount} from 'enzyme';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import Resume from 'Resume';

describe('Resume test', () => {
	it('should renter correctly with no props', () => {
		const component = shallow(<Resume />);
		expect(component).toMatchSnapshot();
		expect(component.state('error')).toBeNull();
	});
});
