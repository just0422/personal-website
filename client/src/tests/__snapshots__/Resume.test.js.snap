// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume test should handle experiences error 1`] = `ReactWrapper {}`;

exports[`Resume test should handle skills error 1`] = `ReactWrapper {}`;

exports[`Resume test should mount correctly 1`] = `ReactWrapper {}`;

exports[`Resume test should mount correctly with no props 1`] = `ReactWrapper {}`;

exports[`Resume test should shallow render correctly 1`] = `ShallowWrapper {}`;
